src/hooks/useChatSocket.tsx

```diff
@@ -72,41 +101,57 @@ export const useChatSocket = () => {
 
     }, [SOCKET_SERVER_URL])
 
-    const setUserName = useCallback((name: string) => {
-        socket?.emit(ClientEvents.INIT_USER, { name });
+    const setUserName = useCallback(async (name: string) => {
+
+        const keyPair = await generateKeyPair();
+        const publicKey = await exportPublicKey(keyPair.publicKey);
+        await storeKey(KeyTransaction.SELF_PK, keyPair.privateKey);
+
+        socket?.emit(ClientEvents.INIT_USER, { name, publicKey });
         setUser(name);
         if (typeof window != 'undefined' && window.localStorage) {
-          localStorage.setItem("username", name);
+            localStorage.setItem("username", name);
         }
     }, [socket]);
 
     const findPartner = useCallback(() => {
         socket?.emit(ClientEvents.FIND_PARTNER);
     }, [socket]);
 
-    const sendMessage = useCallback((message: string | null = null, image: string | null = null, reply:number | null = null) => {
+    const sendMessage = useCallback(async (message: string | null = null, image: string | null = null, reply: number | null = null) => {
         if (socket && partner) {
-            socket.emit(ClientEvents.SEND_MESSAGE, { message, image, reply });
             setMessages(prevMessages => [...prevMessages, { from: 'You', body: message, image, reply }]);
+
+            const partnerPk = await getKey(KeyTransaction.PARTNER_PK);
+            const selfPk = await getKey(KeyTransaction.SELF_PK);
+            const sharedSecret = await deriveSharedSecret(selfPk!!, partnerPk!!);
+            const encryptedMsg = await encryptMessage(message ?? '', sharedSecret);
+
+            if (image){
+                const encryptedImg = await encryptMessage(image, sharedSecret);
+                socket.emit(ClientEvents.SEND_MESSAGE, { message: encryptedMsg, image: encryptedImg, reply });
```

Comment:
There is an edge case where body can be null but image is not null. In these case you will encrypt body containing `null` value only to be not rendered in the frontend.

Reply:
oh no i already took care of that. try sending just an image

Reply:
```suggestion
            const encryptedMsg = await encryptMessage(message ?? '', sharedSecret);
```
This line checks if the message is null

Reply:
That doesn't work I think since I am receiving an encrypted null on the other side anyway. I think that is a wrong operator. You should replace `??` with `||`.